sound/soc/sof/intel/hda-stream.c

```diff
@@ -221,9 +222,10 @@ hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction, u32 flags)
 	/*
 	 * Prevent DMI Link L1 entry for streams that don't support it.
 	 * Workaround to address a known issue with host DMA that results
-	 * in xruns during pause/release in capture scenarios.
+	 * in xruns during pause/release in capture scenarios. This is not needed for the ACE IP.
 	 */
-	if (!(flags & SOF_HDA_STREAM_DMI_L1_COMPATIBLE)) {
+	if (chip_info->hw_ip_version < SOF_INTEL_ACE_1_0 &&
+	    !(flags & SOF_HDA_STREAM_DMI_L1_COMPATIBLE)) {
```

Comment:
hda_dsp_stream_get and put is a generic function, I don't like some hw related code here and it is not related to stream allocation.   why not move them to its caller hda_dsp_pcm_open which also has some DMI-L1 setting ? And we can implement hda_dsp_mtl_pcm_open for mtl without this setting.

Reply:
we do not have a pcm_close op set for any of our platforms. And Im not sure Iwant to add one just for this. What I'd like to do instead is to move the DMI L1 disabling to platform hw_params() and then we can handle all DMI L1 enabling in the hw_free() op. This would actually be better because we won't have to deal with this in the suspend op. What do you think @plbossart ?